src/chapters/unit-testing/mstest.rst

```diff
@@ -0,0 +1,347 @@
+   :alt: WINDOWS: Create MSTest project
+
+   WINDOWS: Creating MSTest project in Visual Studio
+
+**MSTest** is a C# testing framework. When we create a Visual Studio MSTest Project, the 
+necessary API and classes are added as **dependencies** of the ``CarTests`` project. A dependency 
+is a separately developed program or piece of code that another program or piece of code 
+uses to carry it out its function. Our C# tests will *depend* on MSTest code. 
+
+Along the same lines, since ``CarTests`` tests the methods inside of ``Car``, we must add the 
+``Car`` project as a dependency of ``CarTests``.
+
+Right click on on the ``Dependencies`` directory in ``CarTests`` and add a reference to 
+the ``Car`` project.
+
+.. figure:: ./figures/vs-add-dependency-reference.png
+   :alt: Add main project as dependency for test project
+
+   Add main project as dependency for test project
+
+``Car`` and ``CarTests``
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Open the ``Car`` class within ``main`` and look around. Here, we provide a class, ``Car``, with basic 
+information about a make, model, gas level, and mileage. We also give it getters, setters, and a few other methods. 
+
+In the same project, the ``Program`` class contains a main method that prints the
+``make`` and ``model`` of a given ``Car`` object. Run the project to verify it works.
+Now, open ``CarTests``. It's empty, save for a few TODOs. Let's tackle the
+first TODO to make a new empty test. Starting with an empty test lets us validate that we can 
+use MSTest in our current environment.
+
+.. index:: ! test runner
+
+``[TestClass]`` and ``[TestMethod]``
+------------------------------------
+
+Another benefit of coding in an IDE, Visual Studio contains its own **test runner**. A test runner is 
+simply a tool to execute tests and deliver their results. In order to indicate that ``CarTests`` contains
+unit tests that we want the test runner to run, we must give it the ``[TestClass]`` attribute. As you might 
+guess, ``[TestMethod]`` annotates a method to signal it as a test case. Both of these attributes come to us 
+via the Visual Studio test runner.
+
+In ``CarTests``, on top of ``public class CarTests``, add ``[TestClass]``. Then, create the following empty 
+test underneath the first TODO. As usual, be sure write this code rather than copy/paste it:
+
+.. sourcecode:: c#
+   :linenos: 
+
+   using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+   namespace CarTests
+   {
+      [TestClass]
+      public class CarTests
+      {
+         //TODO: add emptyTest so we can configure our runtime environment
+         [TestMethod]
+         public void EmptyTest() {
+            Assert.AreEqual(10,10,.001);
+         }
+         // ,,. other TODOs omitted here
+      }
+   }
+
+Our empty test is aptly named ``EmptyTest()``, a description of its role. This test does 
+not follow the AAA rule from our :ref:`testing-best-practices`, as it jumps straight to 
+asserting. Nor is it relevant, for that matter. 
```

Comment:
The text should answer why this test is important and why it skips the first two A's